core/vm/memory.go

```diff
@@ -43,6 +48,20 @@ func (m *Memory) Set(offset, size uint64, value []byte) {
 	}
 }
 
+// Set32 sets the 32 bytes starting at offset to the value of val, left-padded with zeroes to
+// 32 bytes.
+func (m *Memory) Set32(offset uint64, val *big.Int) {
+	// length of store may never be less than offset + size.
+	// The store should be resized PRIOR to setting the memory
+	if 32 > uint64(len(m.store)) {
```

Comment:
You're validating only the size, not the `offset + size`. Shouldn't you have `offset + 32` here?

Reply:
If you expand the code snippet a bit, you'll see that I reused the code above as much as possible, for easier review. I also thought it should be `offset+32`, but it's not in the code above... (?)

Reply:
In the snippet above, I believe that if `size` is zero, the `offset` can be arbitrarily high. So checking `size+offset` for that would be wrong. So whereas in this instance such a check would be ok, this code is no more wrong than the code above. Both depends on resize already having taken place.